File: shared/src/commonMain/kotlin/com/jooheon/dday/domain/model/HomeUiEvent.kt
package com.jooheon.dday.domain.model

sealed interface HomeUiEvent {
    data object OnAddButtonClick: HomeUiEvent
    data class OnDetailClicked(val model: Dday): HomeUiEvent
    data class OnDelete(val id: Long): HomeUiEvent
    data class OnToggleSelected(val model: Dday): HomeUiEvent
    data class OnProfileImageSelected(val url: String): HomeUiEvent
}

File: iosApp/iosApp/home/model/HomeUiEvent.swift
import Foundation
import PhotosUI
import shared

enum HomeUiEvent {
    case onAddButtonClick
    case onDetailClicked(model: Dday)
    case onDelete(id: Int64)
    case onToggleSelected(model: Dday)
    case onProfileImageSelected(uiImage: UIImage)
}
